api/base/views.py

```diff
@@ -572,7 +572,20 @@ def bulk_get_file_nodes_from_wb_resp(self, files_list):
 
             # mirrors BaseFileNode get_or_create
             try:
-                file_obj = base_class.objects.get(target_object_id=node.id, target_content_type=content_type, _path='/' + attrs['path'].lstrip('/'))
+                if attrs['kind'] == 'file':
+                    file_obj = base_class.objects.get(
+                        target_object_id=node.id,
+                        target_content_type=content_type,
+                        _path='/' + attrs['path'].lstrip('/'),
+                        modified=attrs['modified'],
```

Comment:
I agree adding this `modified` constraint will solve the majority of the issues with /v2/.